src/wallet/rpcwallet.cpp

```diff
@@ -1518,31 +1530,37 @@ UniValue listtransactions(const UniValue& params, bool fHelp)
     if(params.size() > 3)
         if(params[3].get_bool())
             filter = filter | ISMINE_WATCH_ONLY;
-
+    string address("*");
+    if (params.size()>4)
+    {
+        address=params[4].get_str();
+        if (address!=("*"))
+        {
+            CBitcoinAddress baddress = CBitcoinAddress(address);
+            if (!baddress.IsValid())
+                throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid Zen address");
+        }
+    }
     if (nCount < 0)
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Negative count");
     if (nFrom < 0)
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Negative from");
 
-    UniValue ret(UniValue::VARR);
+    std::multimap<int64_t, CWalletTx > txes;
```

Comment:
Restore the previous serialization format and modify OrderedTxItems to allow filtering per address